scripts/vehicle_node.py

```diff
@@ -0,0 +1,371 @@
+        rospy.spin()
+
+    def create_terrain_map(self):
+
+        #Get parameters
+        mu = rospy.get_param('/mu')
+        sigma = rospy.get_param('/sigma')
+        sigma_filt = rospy.get_param('/sigma_filt')
+        height = rospy.get_param('/height')
+        width = rospy.get_param('/width')
+        resolution =  rospy.get_param('/resolution')
+
+        #Fill array of size h x w with Gaussian Noise given the height and width of the point cloud
+        h = int(((height*2) + 1)*(resolution*1000))
+        w = int(((width*2) + 1)*(resolution*1000))
+        grid_map = (h,w)
+        gaussian_array = np.random.normal(mu, sigma, grid_map)
+
+        #Filter the array to smoothen the variation of the noise
+        gaussian_array = gaussian_filter(gaussian_array, sigma_filt)
+
+        #Create an list and fill the list with x, y values and z the values will be the filtered gaussian noise
+        # TODO perform these calculations without for loops
+        incr_x = 0
+        incr_y = 0
+        cloud = []
+        for x in np.arange(-height, (height+1), resolution):
+            for y in np.arange(-width, (width+1), resolution):
+                innerlist = []
+                innerlist.append(x)
+                innerlist.append(y)
+                innerlist.append(gaussian_array[incr_x][incr_y])
+                cloud.append(innerlist)
+                incr_y+=1
+            incr_x+= 1
+            incr_y = 0
+
+        return cloud
+
+
+    def velocity_cmd_callback(self, data):
+
+        #assign updated command velocities to twist variable
+
+        self.twist = data
+
+
+    def update_position(self, event):
+
+        #Load parameters
+        resolution =  rospy.get_param('/resolution')
+        v_height =  rospy.get_param('/v_height')
+        v_width =  rospy.get_param('/v_width')
+
+        #Local variable to calculate position based on most recent command velocity data
+        position_data = self.twist
+
+        #time elapsed since last update position call
+        if hasattr(event, 'last_real'):
+            if event.last_real is None:
+                event.last_real = rospy.Time.now()
+                time = event.current_real - event.last_real
+            else:
+                time = event.current_real - event.last_real
+        
+        time = time.to_sec()
+
+        #Calculate angle turned in the given time using omega = theta/time
+        angle = position_data.angular.z*time
+
+        #Calculate distance travelled in the given time using linear velocity = arc distance/time
+        distance = position_data.linear.x*time
+
+        #Calculate yaw, pitch and roll of the robot (pitch and roll currently not calculated)
+        self.vehicle_roll = 0.0
+        self.vehicle_pitch = 0.0
+        self.vehicle_yaw += angle
+
+        #Calculate vehicle x, y, z position coordinates 
+        self.vehicle_x += (distance)*cos(self.vehicle_yaw)
+        self.vehicle_y += (distance)*sin(self.vehicle_yaw)
```

Comment:
This trig calculation assumes the following:
1) Yaw the vehicle by `angle`
2) After completing the yaw, drive straight for `distance`

In actuality, the yaw and translation happen at the same time.  I'd suggest taking a little time to do some calculus and see what the final position and orientation will look like if the yaw and translation happen simultaneously.

Or, add a TODO to revisit this later.